src/pages/search/components/search-form.jsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import './search-form.css'
+
+import { searchBy } from '../../../enums'
+
+import { Radio } from '../../../components'
+
+export class SeachForm extends React.PureComponent {
+  static propTypes = {
+    search: PropTypes.string.isRequired,
+    searchBy: PropTypes.string.isRequired,
+    onSearchChange: PropTypes.func.isRequired
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      search: this.props.search,
+      searchBy: this.props.searchBy
```

Comment:
I know I read [about it](https://reactjs.org/docs/react-component.html#constructor).
I didn't know an other way to pass changed data on form submit. I don't want to pass `search` and `searchBy` values via separate callbacks. Also it's supposed to be onetime initialization.

On second thought, I think I can override `getDerivedStateFromProps`, so that component state is in sync with props. Will it work here?

Reply:
Good article and definition state as "forks of props" for this case, I haven't been read this docs a long time.
Why do you think parent component won't want to change props? 
To be honest I don't understand why do you you local state in the component. I think it's enough to use [uncontrolled component](https://reactjs.org/docs/uncontrolled-components.html) without state. What do you think?

Reply:
I see search params data flow as:
1. The page component is loaded - it get's initial search params from URL query string
2. The component passes search parameters to it's children components
3. Child components set their inputs values from props
4. If a component contains a form element it accumulates inputs' data and sends data to parent on submit
5. In general, child components sends data to parent when data changed via user input
6. The page component updates URL query string and update

I think the parent shouldn't change search params to avoid erasing user's input

Using uncontrolled components looks cumbersome to me. But I will ty to implement logic this way